src/java/org/apache/cassandra/dht/AbstractBounds.java

```diff
@@ -51,6 +51,7 @@ private enum Type
     public AbstractBounds(T left, T right)
     {
         assert left.getPartitioner() == right.getPartitioner();
+        assert left.compareTo(right) <= 0 : "left " + left + " > right " + right;
```

Comment:
Given this method, this assertion is invalid:

https://github.com/datastax/cassandra/blob/2bdda4c4f07e23d8f1370c33f2d159e00ce27f6a/src/java/org/apache/cassandra/dht/Range.java#L316-L322